src/main/scala/CloudWatchMetricFilter.scala

```diff
@@ -0,0 +1,15 @@
+
+import com.typesafe.scalalogging.LazyLogging
+import com.codahale.metrics.{Metric, MetricFilter}
+import scala.util.matching.Regex
+
+object CloudWatchMetricFilter extends LazyLogging {
+    def buildMetricFilter(filterRegexPattern: Regex): MetricFilter = {
+        return new MetricFilter() {
+            override def matches(string: String, metric: Metric): Boolean =  {
+                return filterRegexPattern.findFirstIn(string).nonEmpty
```

Comment:
`return` isn't needed here or above

Reply:
sorry its actually the first time coding in scala (have done java)
so should it be like

```
def buildMetricFilter(filterRegexPattern: Regex): MetricFilter = {
        new MetricFilter() {
            override def matches(string: String, metric: Metric): Boolean =  filterRegexPattern.findFirstIn(string).nonEmpty
       }
```

doesnt matches need a return value of a boolean and buildmetricfilter to have a return of a MetricFilter?

Reply:
Scala automatically returns the value of the last statement for you so you don't have to use the 'return' keyword

Reply:
wait thats pretty cool! Thanks for the tip!

Reply:
I think this can be simplified to:
```
class CloudWatchMetricFilter(filterRegexPattern: Regex) extends MetricFilter {
  override def matches(string: String, metric: Metric): Boolean =  {
    filterRegexPattern.findFirstIn(string).nonEmpty
  }
}```

Reply:
scala is interesting 👍 :')

Reply:
Can you update the object to be a class instead as shown above?